Source/DataSources/KmlDataSource.js

```diff
@@ -255,6 +255,23 @@ define([
         return deferred.promise;
     }
 
+    function insertNamespaces(text) {
+        var namespaceMap = {
+            'xsi:' : ['xmlns:xsi=', '"http://www.w3.org/2001/XMLSchema-instance"']
+        };
+        var firstPart;
+        var lastPart;
+
+        for (var key in namespaceMap){
+            if(text.indexOf(key) !== -1 && text.indexOf(namespaceMap[key][0]) === -1) {
+                firstPart = text.substr(0, text.indexOf('<kml') + 4);
```

Comment:
You can find the index of `<kml` once before the loop, then maintain the insert index as you iterate `namespaceMap` and possibly add a few schemas.

Reply:
You still only want to find `<kml` once, but you may want to do it in the loop after we found a problem, so we don't search a potentially long string if we don't need to. So something like this

```
var firstPart;
var lastPart;
for (var key in namespaceMap){
...
  if(text.indexOf(prefix) !== -1 && text.indexOf(declaration) === -1) {
    if (!defined(firstPart)) {
      // Search for <kml and set firstPart/lastPart
    }
    ...
  }
}

if (defined(firstPart)) {
  text = firstPart + lastPart;
}

return text;
```